### PR TITLE
[React Components] update setShippingAddressOnCart to refetch cart details

### LINKED ISSUE #269: Checkout issue - tax not being added
### Expected Behaviour
Tax is added to grand total after a shipping address has been set on the cart.
### Actual Behaviour
Tax is missing until page is refreshed and cart query is run.
### Reproduce Scenario (including but not limited to)
Checkout as a guest or logged in user
#### Steps to Reproduce
1. Add item to cart.
2. Click on checkout button.
3. Open shipping address view, enter address, click use address button.
4. Tax is missing from total.
#### Platform and Version
AEM 6.5
CIF Core Components 0.10.1
Magento 2.3.4

### PR DESCRIPTION
**Pull request by dabe5651**
## Description

Updated checkout shipping address to rerfetch cart details query now that we have address to use for taxes to fix missing tax from grand total. 

## Related Issue

Fixes #269. 

## Motivation and Context

This is required so users aren't able to checkout without taxes being applied to the cart.

## How Has This Been Tested?

I tried adding an address as a guest and signed in users and saw the tax apply correctly.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: react-components/src/components/Checkout/editableForm.js [hunk 1]
```diff
@@ -64,7 +68,15 @@ const EditableForm = props => {
 
     const handleSubmitAddressForm = useCallback(
         formValues => {
-            setShippingAddressesOnCart({ variables: { cartId: cartId, countryCode: 'US', ...formValues } });
+            setShippingAddressesOnCart({ variables: { cartId: cartId, countryCode: 'US', ...formValues } })
+                .catch(error => {
+                    dispatch({ type: 'error', error: error.toString() });
+                })
+                .finally(() => {
+                    dispatch({ type: 'endLoading' });
```
Comment (herzog31) [inline react-components/src/components/Checkout/editableForm.js:76]: The `beginLoading` and `endLoading` events are only available via the `CartContext`, so you need to use `cartDispatch` here.

Comment (dabe5651) [reply react-components/src/components/Checkout/editableForm.js]: Thanks, I updated the events to use cartDispatch here.

Comment (herzog31) [reply react-components/src/components/Checkout/editableForm.js]: @dabe5651 please make sure you run tests locally first and they pass. CircleCI reports test failures.

**Comment by herzog31** [comment]
Somehow CircleCI doesn't like this fork. But I checked out the branch locally and confirmed that all tests pass.

**Comment by dplaton** [comment]
I checked the CI pipeline with another branch, it worked. I'll merge the PR.
Thanks for the contribution, @dabe5651 !

**Pull request merged**